mercantile/__init__.py

```diff
@@ -61,6 +61,47 @@ class InvalidLatitudeError(MercantileError):
     """Raised when math errors occur beyond ~85 degrees N or S"""
 
 
+class InvalidZoomError(MercantileError):
+    """Raised when a zoom level is invalid"""
+
+
+class ParentTileError(MercantileError):
+    """Raised when a parent tile cannot be determined"""
+
+
+class QuadKeyError(MercantileError):
+    """Raised when errors occur in computing or parsing quad keys"""
+
+
+class TileArgParsingError(MercantileError):
+    """Raised when errors occur in parsing a function's tile arg(s)"""
+
+
+def _parse_tile_arg(*args):
+    """parse the *tile arg of module functions
+
+    Parameters
+    ----------
+    tile : Tile or sequence of int
+        May be be either an instance of Tile or 3 ints, X, Y, Z.
+
+    Returns
+    -------
+    Tile
+
+    Raises
+    ------
+    TileArgParsingError
+
+    """
+    if len(args) == 1:
+        args = args[0]
+    if len(args) == 3:
+        return Tile(*args)
+    else:
+        raise TileArgParsingError("the tile argument may have 1 or 3 values. Note that zoom is a keyword-only argument")
```

Comment:
Here's the new argument parsing exception in use.

Reply:
Great!